dockerHashing/src/main/scala/cromwell/docker/DockerImageIdentifier.scala

```diff
@@ -38,28 +36,35 @@ object DockerImageIdentifier {
        )                                        # End capturing group for name
        
        (?:   
-          (                                     # Begin capturing group for tag separator
-            :|@                                 # Tag separator. ':' is followed by a tag, '@' is followed by a digest
-          )                                     # End capturing group for tag separator
+          :                                     # Tag separator. ':' is followed by a tag
        
           (                                     # Begin capturing group for reference 
             [A-Za-z0-9]+(?:[.:_-][A-Za-z0-9]+)* # Reference
           )                                     # End capturing group for reference  
-       )?                                      
+       )?
+       (?:   
+          @                                     # Tag separator '@' is followed by a digest
+             
+          (                                     # Begin capturing group for reference 
+            [A-Za-z0-9]+(?:[.:_-][A-Za-z0-9]+)* # Reference
+          )                                     # End capturing group for reference  
+       )?
        """.trim.r
   
   def fromString(dockerString: String): Try[DockerImageIdentifier] = {
     dockerString match {
         // Just a name means latest tag implicitly
-      case DockerStringRegex(name, null, null) => buildId(name, TagSeparator,  DefaultDockerTag)
-      case DockerStringRegex(name, tagSeparator, reference) => buildId(name, tagSeparator, reference)
+      case DockerStringRegex(name, null, null) => buildId(name, None, None)
+      case DockerStringRegex(name, tag, null) => buildId(name, Option(tag), None)
+      case DockerStringRegex(name, null, hash) => buildId(name, None, Option(hash))
+      case DockerStringRegex(name, tag, hash) => buildId(name, Option(tag), Option(hash))
```

Comment:
Since `Option(null)` is already `None`, can we remove the other cases and just use this one for everything?
```
case DockerStringRegex(name, tag, hash) => buildId(name, Option(tag), Option(hash))
```